applications/googlehome-meets-dotnetcontainers/GoogleHomeAspNetCoreDemoServer/Dialogflow/Intents/BigQuery/BigQueryNoaaextremeHandler.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright(c) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+using Google.Cloud.BigQuery.V2;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GoogleHomeAspNetCoreDemoServer.Dialogflow.Intents.BigQuery
+{
+    /// <summary>
+    /// Handler for all "bigquery.noaaextreme" DialogFlow intent.
+    /// </summary>
+    [Intent("bigquery.noaaextreme")]
+
+    public class BigQueryNoaaextremeHandler : BaseBigQueryHandler
+    {
+        /// <summary>
+        /// Initializes class with the conversation.
+        /// </summary>
+        /// <param name="conversation">Conversation</param>
+        public BigQueryNoaaextremeHandler(Conversation conversation) : base(conversation)
+        {
+        }
+
+        /// <summary>
+        /// Handle the intent.
+        /// </summary>
+        /// <param name="req"></param>
+        /// <returns></returns>
+        public override async Task<object> Handle(ConvRequest req)
+        {
+            var errorMessage = ExtractAndValidateParameters(req, out string temp, out string year, out string countryCode2, 
+                out string countryName, out string fipsCountry);
+
+            if (errorMessage != null) return DialogflowApp.Tell(errorMessage);
+
+            var bigQueryClient = await BigQueryClient.CreateAsync(Program.AppSettings.GoogleCloudSettings.ProjectId);
+
+            // Build the parameterized SQL query
+            var tableGsod = bigQueryClient.GetTable("bigquery-public-data", "noaa_gsod", $"gsod*");
+            var tableStations = bigQueryClient.GetTable("bigquery-public-data", "noaa_gsod", "stations");
+            var sql = $"SELECT (gsod.temp - 32)*5/8 AS celcius, stations.name AS name, gsod.year AS y, gsod.mo AS m, gsod.da AS d \n" +
+                $"FROM {tableGsod} AS gsod \n" +
```

Comment:
Is there any potential for an SQL injection attack here?

Reply:
Nothing to change here then.